### PR TITLE
Add feature to export env variables

### LINKED ISSUE #23: Pass environment variables into Docker
I want to set `OPAMWITHTEST=true` to perform [`run-test`](https://opam.ocaml.org/doc/Manual.html#opamfield-run-test), but cannot do it with [GitHub Actions' `env`](https://docs.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables#about-environment-variables), because that parameter should be passed into Docker.

### PR DESCRIPTION
**Pull request by erikmd**
- **Kind:** feature
- Close #23 (this feature being requested by @liyishuai)
- **Depends-on: #32 (to be merged before #31)**

FYI I've done a system test, relying on Yishuai's suggestion to pass `OPAMWITHTEST`:
- https://github.com/erikmd/docker-coq-github-action-demo/blob/337ecf043c94303b720062069f99d8a8d4e52555/.github/workflows/build-coq-demo.yml#L39-L54

and it seems to work pretty well:
~- https://github.com/erikmd/docker-coq-github-action-demo/runs/1456333460?check_suite_focus=true~ [see [comment below](https://github.com/coq-community/docker-coq-action/pull/31#issuecomment-734007307)]

As an aside, I didn't implement that other suggestion for the moment:
- https://github.com/coq-community/docker-coq-action/issues/23#issuecomment-657271610

but it could just as well be passed using the feature of this PR #31 (unless we decide to hard-code `OPAMYES` anyway!)

Maybe this can be merged on tomorrow Thursday, after #30 and #32, as soon as one of you (@Zimmi48, @liyishuai) took a look.

### CODE REVIEW

**Comment by erikmd** [comment]
The shellcheck linting test is OK now.
Also, I'll run another test incorporating the #32 patch.

Latest test done: cf. https://github.com/erikmd/docker-coq-github-action-demo/runs/1456454613?check_suite_focus=true FYI

**Review by liyishuai — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -171,6 +171,39 @@ steps:
...
+        printf "%s\n" "$first_example"
+        startGroup Build dependencies, with-test
+          opam pin add -n -y -k path $PACKAGE $WORKDIR
+          opam update -y
+          opam install -y -j 2 $PACKAGE --deps-only
+        endGroup
+      export: 'first_example OPAMWITHTEST'
+    env:
+      first_example: 'some value'
+      OPAMWITHTEST: 'true'
```
Comment (Zimmi48) [inline README.md]: I would not put a `custom_script` in this example. Only show the example of `OPAMWITHTEST` and explain why it's useful.

Comment (erikmd) [reply README.md]: OK I'll refactor this

Comment (erikmd) [reply README.md]: @Zimmi48 done, adding two docs hyperlinks as well

**Comment by erikmd** [comment]
@Zimmi48 I propose to merge this no later than tomorrow Friday noon

**Review by Zimmi48 — APPROVED** [review]

**Comment by Zimmi48** [comment]
BTW, please do not @-mention me in commit messages because I get a notification every time these commits are rebased.

**Pull request merged**